### ISSUE TITLE
transformer 4.45 -> causing  image_to_text failures for gaudi attribute error

### ISSUE THREAD

**Issue by jiminha** [huggingface/optimum-habana#1357]
### System Info

```shell
Gaudi2/ Gaudi3 
Docker 1.18.0
```

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

GAUDI2_CI=1  RUN_SLOW=1 python -m pytest tests/test_image_to_text_example.py -s -v -k llava-1.5-7b-hf

```
Traceback (most recent call last):
  File "/root/tf/optimum-habana-f/examples/image-to-text/run_pipeline.py", line 218, in <module>
    main()
  File "/root/tf/optimum-habana-f/examples/image-to-text/run_pipeline.py", line 152, in main
    generator = pipeline(
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/__init__.py", line 895, in pipeline
    framework, model = infer_framework_load_model(
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/base.py", line 287, in infer_framework_load_model
    model = model_class.from_pretrained(model, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/transformers/models/auto/auto_factory.py", line 557, in from_pretrained
    return model_class.from_pretrained(
  File "/usr/local/lib/python3.10/dist-packages/transformers/modeling_utils.py", line 3848, in from_pretrained
    model = cls(config, *model_args, **model_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/transformers/models/llava/modeling_llava.py", line 245, in __init__
    super().__init__(config)
  File "/usr/local/lib/python3.10/dist-packages/transformers/modeling_utils.py", line 1403, in __init__
    self.generation_config = GenerationConfig.from_model_config(config) if self.can_generate() else None
  File "/usr/local/lib/python3.10/dist-packages/transformers/generation/configuration_utils.py", line 1239, in from_model_config
    is_unset = getattr(generation_config, attr) == getattr(default_generation_config, attr)
AttributeError: 'GenerationConfig' object has no attribute 'trim_logits'
```

### Expected behavior

We are getting attribute error on the transformer_future release for all image_to_text tests. (all 10 tests)
 AttributeError: 'GenerationConfig' object has no attribute 'trim_logits'
 
This is caused by he Transformer change as below,  getattr(config, attr) should be called after checking if attr is in decoder_config just as it was in 4.43 release. Here the config is GaudiGenerationConfig  which includes gaudi specific attributes, so it fails to read from the default_generation_config in v4.45. I'm assuming other vendors who have  different generation config can also extend this configurations? 
Can we have this code back to similar to 4.43.4 instead of having is_unset variable?   

https://github.com/huggingface/transformers/blob/v4.43.4/src/transformers/generation/configuration_utils.py#L1182
```
    if attr in decoder_config and getattr(config, attr) == getattr(default_generation_config, attr):
                        setattr(config, attr, decoder_config[attr])

```

https://github.com/huggingface/transformers/blob/main/src/transformers/generation/configuration_utils.py#L1245C72-L1245C97
```
is_unset = getattr(generation_config, attr) == getattr(default_generation_config, attr)
                if attr in decoder_config_dict and is_unset:
                    setattr(generation_config, attr, decoder_config_dict[attr])
```

@regisss  please advise on this, otherwise we will need to override this function.

**Comment by jiminha** [comment]
@regisss  could you check this issue and comment on my question? Not sure if we should just override this, or the fix should come from transformer.

**Comment by regisss** [comment]
Yes, I'm just waiting for the next Transformers release that should happen in 1-2 hours and then I'll check the issues you posted

**Comment by regisss** [comment]
This issue is fixed now in https://github.com/huggingface/optimum-habana/pull/1359.
But other issues appeared for Llava and Llava-next.

**Comment by imangohari1** [comment]
> This issue is fixed now in #1359. But other issues appeared for Llava and Llava-next.

@regisss 
Thanks for the update. The GenerationConfig error is now fixed with #1359 but we are seeing some unexpected behaviour with llava on img2text 
```
python3 examples/image-to-text/run_pipeline.py  --model_name_or_path llava-hf/llava-1.5-7b-hf  --use_hpu_graphs  --bf16
```
leads to no output (and therefore negative throughput)
```
09/25/2024 22:48:02 - INFO - __main__ - result = [[{'generated_text': 'USER:  \nWhat is shown in this image? ASSISTANT:'}]]
09/25/2024 22:48:02 - INFO - __main__ - time = 4410.16705436632ms, Throughput (including tokenization) = -0.4534975603746993 tokens/second
```
on the main the output is like this:
```
09/26/2024 00:40:19 - INFO - __main__ - result = [[{'generated_text': 'USER:  \nWhat is shown in this image? ASSISTANT: The image shows a pier extending out over a body of water, with a mountain in the background. The pier is made of wood and appears to be a popular spot for people to gather and enjoy the view.'}]]
09/26/2024 00:40:19 - INFO - __main__ - time = 1914.556894917041ms, Throughput (including tokenization) = 20.892562715788724 tokens/second
```
Do you see the same issue on your end? Any tips on how we go about fixing this? 
Thanks.

**Comment by regisss** [comment]
Yes, I'm aware, that's what I was referring to above. Both Llava and Llava-next have (different) issues.
I need to investigate to know more.

**Comment by regisss** [comment]
@jiminha @imangohari1 The last commit I pushed to #1359 should fix these issues: https://github.com/huggingface/optimum-habana/pull/1359/commits/8043d2cef69edc9eae6c7282bbb7fa41f268e5b6
Let me know if it works on your side too.

It's a workaround, I'll see later how to have a proper fix. It comes from these two lines that were recently added in Transformers: https://github.com/huggingface/optimum-habana/blob/8043d2cef69edc9eae6c7282bbb7fa41f268e5b6/optimum/habana/transformers/generation/utils.py#L1149-L1150

**Comment by imangohari1** [comment]
> @jiminha @imangohari1 The last commit I pushed to #1359 should fix these issues: [8043d2c](https://github.com/huggingface/optimum-habana/commit/8043d2cef69edc9eae6c7282bbb7fa41f268e5b6) Let me know if it works on your side too.
> 
> It's a workaround, I'll see later how to have a proper fix. It comes from these two lines that were recently added in Transformers:
> 
> https://github.com/huggingface/optimum-habana/blob/8043d2cef69edc9eae6c7282bbb7fa41f268e5b6/optimum/habana/transformers/generation/utils.py#L1149-L1150

Thank you. The WA appears to getting pass the output generation issue 👍
